doc/ExpertOptions.md

```diff
@@ -1441,13 +1441,23 @@ For an in depth discussion, please see [Missing Call Arguments](MissingCallArgum
 
 # Manifests
 
-In extreme cases, running compiled workflows can fail due to DNAnexus platform limits on the total size of the input and output JSON documents of a job. An example is a task with many inputs/outputs that is called in scatter over a large collection. In such a case, you can enable manifest support at compile time with the `-useManifests` option. This option causes each generated applet or workflow to accept inputs as a manifest, and to produce outputs as a manifest.
-
-A manifest is a JSON document that contains all the inputs/outputs that would otherwise be passed directly to/from the applet. A manifest can be specified in one of two ways: via a JSON input, or via a File input (where the file must exist on the platform).
+In extreme cases, running compiled workflows can fail due to DNAnexus platform limits on the total size of the input and 
+output JSON documents of a job. An example is a task with many inputs/outputs that is called in scatter over a large collection. 
+In such a case, you can enable manifest support at compile time with the `-useManifests` option. 
+This option causes each generated applet or workflow to accept inputs as a manifest, and to produce outputs as a manifest.
+
+A manifest is a JSON document that contains all the inputs/outputs that would otherwise be passed directly to/from the 
+workflow stage. A manifest can be specified in one of two ways: 
+1. A `.json` input file (see `Manifest JSON`). Here, when the workflow is **compiled** with `-inputs mymanifest.json`, a new 
+file `mymanifest.dx.json` will be generated locally. This is a recommended way to provide inputs in the manifest format.
+2. A `file-xxx` input, where the file must exist on the platform (see `Intermediate manifest file inputs/outputs`). A 
+typical use case for this scenario is when a user wants to pass manifest output file from a stage (including `output` stage) 
+directly to a new workflow. Also, this scenario might be useful when debugging individual stages of a failing workflow.
 
 ## Manifest JSON
 
-When manifest support is enabled, each applet has an `input_mainfest___` input field of type `hash`, which means that it accepts a JSON document as a string. For example, given the following workflow:
+When manifest support is enabled, each applet has an `input_mainfest___` input field of type `hash`, which means that it 
+accepts a JSON document as a string or as a `.json` file. For example, given the following workflow:
```

Comment:
I though dnanexus applets input of type hash expects a json hash.    Could you give me an example of passing 1. json document as a string 2. .json file

Reply:
From the `dx run` documentation: 
```
-f FILENAME, --input-json-file FILENAME
                        Load input JSON from FILENAME ("-" to use stdin)
```
example of the json file is in the text (referenced in another comment). Successful run with `.json` file (document) - `project-Fy9QqgQ0yzZbg9KXKP4Jz6Yqanalysis-GFBqG8j0yzZbjGyp159b46VK`.
With `cat mymanifest.dx.json | dx run workflow-xxx -f -`: `project-Fy9QqgQ0yzZbg9KXKP4Jz6Yq:analysis-GFBqgQj0yzZb76p69f7qFkZB`